src/Processors/Transforms/AggregatingTransform.h

```diff
@@ -70,6 +70,21 @@ struct ManyAggregatedData
         for (auto & mut : mutexes)
             mut = std::make_unique<std::mutex>();
     }
+
+    ~ManyAggregatedData()
+    {
+        // Aggregation states destruction may be very time-consuming.
+        // In the case of a query with LIMIT, most states won't be destroyed during conversion to blocks.
+        // Without the following code, they would be destroyed in the destructor of AggregatedDataVariants in the current thread (i.e. sequentially).
+        const auto pool = std::make_unique<ThreadPool>(variants.size());
+
+        for (auto && variant : variants)
+        {
+            // It doesn't make sense to spawn a thread if the variant is not going to actually destroy anything.
+            if (variant->aggregator)
+                pool->trySchedule([variant = std::move(variant)]() {});
```

Comment:
Maybe add a comment like:
"variant is moved here and will be destroyed in the destructor of the lambda function".

By the way, does ThreadPool run the destructor of the lambda function in a thread?

Reply:
> By the way, does ThreadPool run the destructor of the lambda function in a thread?

seems like that, otherwise, we wouldn't see any change in perf